tests/dogfood.rs

```diff
@@ -13,18 +13,50 @@ fn dogfood() {
     if option_env!("RUSTC_TEST_SUITE").is_some() || cfg!(windows) {
         return;
     }
-    let root_dir = std::env::current_dir().unwrap();
-    for d in &[".", "clippy_lints", "rustc_tools_util", "clippy_dev"] {
+    let root_dir = std::path::PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    let clippy_cmd = std::path::Path::new(&root_dir).join("target/debug/cargo-clippy");
```

Comment:
`cargo run` ensured that the executable was built. I'm not sure if this is a probably but won't this break if the tests are run in release mode without building in debug first i.e. `cargo clean; cargo test --release;`. Also will the `/` path separators work on Windows?

Reply:
That's a good point, is there a good way to handle release vs debug mode in the binary itself?